modem: modules: remove EXPERIMENTAL

bjarki-andreasen opened the pull request:
Remove the EXPERIMENTAL label from the MODEM_MODULES. These modules are fully tested and widely used, and have been present for 2 years.

bjarki-andreasen-nordic committed:
modem: modules: remove EXPERIMENTAL

Remove the EXPERIMENTAL label from the MODEM_MODULES and update
API version to 1.0.0. These modules are fully tested and widely
used, and have been present for 2 years.

Signed-off-by: Bjarki Arge Andreasen <bjarki.andreasen@nordicsemi.no>

ubieda reviewed (commented):
Should we add a version to the APIs under `include/zephyr/modem`?

bjarki-andreasen commented:
> Should we add a version to the APIs under `include/zephyr/modem`?

We can, but I don't think we track the APIs of subsystems :)

kartben commented:
> Its been a long time since I added that :D I just looked at headers in tree to find many many which don't mention a version :)

:)
Yeah I think it would make sense for all "sub"-headers to also have an explicit version set btw

bjarki-andreasen-nordic committed:
modem: add versions to all headers

Versions for specific modem modules don't include versions in their
headers, and the chat module did not even define a group.

This commits updates most modem modules to include versions in
their headers. Specifically, the following modules have their
versions set to 1.0.0, as these should no longer be marked
experimental:

- modem_chat
- modem_cmux
- modem_pipe
- modem_pipelink
- modem_ppp
- modem_ubx

Though the modem_ubx was recently refactored, it is tested in CI
and is implemented by two drivers, and has been present since
release 3.7, so it is not marked as experimental.

Signed-off-by: Bjarki Arge Andreasen <bjarki.andreasen@nordicsemi.no>

tomi-font reviewed (commented):
4.3, not 4.2?

kartben commented:
> 4.3, not 4.2?

Neither -- the @since should represent when the API was introduced so shouldn't be updated

tomi-font commented on include/zephyr/modem/ubx.h:
```diff
@@ -23,6 +23,8 @@ extern "C" {
 /**
  * @brief Modem Ubx
  * @defgroup modem_ubx Modem Ubx
+ * @since 4.2
```
This doesn't seem to be from 4.2?

bjarki-andreasen replied:
corrected to v3.6
